src/dmd/dtoh.d

```diff
@@ -2881,6 +2881,37 @@ const(char*) keywordClass(const Identifier ident)
         case "xor_eq":
             return "special operator in C++";
 
+        // C keywords
+        case "_Bool":
+        case "_Complex":
+        case "_Imaginary":
+
+        // Common C extensions
+        case "__alignof__ ":
+        case "__asm":
+        case "__asm__":
+        case "__attribute__":
+        case "__cdecl":
+        case "__complex":
+        case "__complex__":
+        case "__const":
+        case "__const__":
+        case "__declspec":
+        case "__func__":
+        case "__has_attribute":
+        case "__imag__":
+        case "__inline":
+        case "__inline__":
+        case "__real__":
+        case "__restrict":
+        case "__restrict__":
+        case "__signed":
+        case "__signed__":
+        case "__typeof":
+        case "__typeof__":
+        case "__volatile":
+        case "__volatile__":
```

Comment:
This entire list could be replaced with a check for anything that starts with `__`.  In C/C++, it's all generally fair game for the compiler to claim any such symbol as having a special meaning (i.e: `__foo`)

Reply:
Good idea, that will also work for many future extensions.